### ISSUE TITLE
Respect parent route query params

### ISSUE THREAD

**Issue by Panman8201** [intercom/ember-href-to#62]
In my app I have a query param at the `application` level. Most of the time it'll never make it's way into the URL because the default value is not changed. However, when it is, it seems like any `{{href-to}}` URL's don't know about that, probably because it isn't passed in as `(query-params)`. The issue would be that child routes don't really have direct access to the current value _to_ pass in. Is there a way for this helper to know about those and add them to the URL automatically?

**Comment by Panman8201** [comment]
@cibernox Would PR #47 resolve this issue?

**Comment by cibernox** [comment]
Yes, that was the idea, but I have to admit that I hit a dead end with that. The logic around query params was quite messy. I should try to finish it, but I don't think I'll have time in the next days.

**Comment by homersimpsons** [comment]
Any news on this ?
I found this thread: https://stackoverflow.com/questions/37268480/how-can-i-get-url-params-in-emberjs-component

**Comment by cibernox** [comment]
@homersimpsons I didn't find a solution for the PR and i forgot about it. I'm not sure if with the new router service almost done in newer versions we should keep pursuing this as it will be an "official" href-to soon.

**Comment by homersimpsons** [comment]
@cibernox Okay, thank you for your answer :)

**Comment by MartinMalinda** [comment]
It should be possible to achieve this via `ember-router-helpers`.
```hbs
{{#with (route-params 'parent.child') as |routeParams|}}
   <a href={{routeParams.url}}
    class={{if routeParams.isActive "active" "inactive"}}
    {{action routeParams.transitionTo}}>Blah</a>
{{/with}}
```

**Comment by patocallaghan** [comment]
The ember-href-to package is now maintained by the Ember community and lives at https://github.com/adopted-ember-addons/ember-href-to. We're closing this issue and archiving this repo. If you still think it's relevant, please reopen it against the new repo.
